app/controllers/topics_controller.rb

```diff
@@ -8,6 +8,38 @@ def index
     end
   end
 
+  def vote
+    @topic = Topic.find params[:id]
+    @topic.increment :votes
+    respond_to do |format|
+      if @topic.save 
```

Comment:
I added a validation for votes, so they must be greater or equal to 0. However since  `@topic.inclement!` skips validations i had to do it in two steps `@topic.increment` and then `@topic.save`